src/families/cosmos/react.js

```diff
@@ -14,3 +24,143 @@ export function useCosmosPreloadData() {
   }, []);
   return state;
 }
+
+export function useCosmosFormattedDelegations(
+  account: Account,
+  option?: FormatOption
+): CosmosFormattedDelegation[] {
+  const { validators } = useCosmosPreloadData();
+  const delegations = account.cosmosResources?.delegations;
+  invariant(delegations, "cosmos: delegations is required");
+
+  const formattedDelegations = useMemo(
+    () => formatDelegations(delegations, validators),
+    [delegations, validators]
+  );
+  const unit = useMemo(() => getAccountUnit(account), [account]);
+
+  switch (option) {
+    case "claimReward":
+      return formattedDelegations
+        .filter(({ pendingRewards }) => pendingRewards.gt(0))
+        .map(({ pendingRewards, ...rest }) => ({
+          ...rest,
+          pendingRewards,
+          reward: formatCurrencyUnit(unit, pendingRewards, {
+            disableRounding: true,
+            alwaysShowSign: false,
+            showCode: true,
+          }),
+        }));
+    case "redelegate":
+    case "undelegate":
+      return formattedDelegations.map((d) => ({
+        ...d,
+        formattedAmount: formatCurrencyUnit(unit, d.amount, {
+          disableRounding: true,
+          alwaysShowSign: false,
+          showCode: true,
+        }),
+      }));
+    default:
+      return formattedDelegations;
+  }
+}
+
+export function useCosmosDelegationsQuerySelector(
+  account: Account,
+  transaction: Transaction,
+  option?: FormatOption
+) {
+  const [query, setQuery] = useState<string>("");
+  const delegations = useCosmosFormattedDelegations(account, option);
+
+  const options = useMemo<CosmosFormattedDelegation[]>(
+    () =>
+      delegations.filter(
+        // [TODO] better query test
+        ({ validator }) =>
+          !query || !validator || new RegExp(query, "gi").test(validator.name)
```

Comment:
if we don't know exactly what this is, we better should expose it as a function (filtering function) rather than accepting obscuring regexp the userland can set 🤔 

i wonder if there is existing engine we have somewhere (fuzzy search, etc..). what did we do for tron?

Reply:
on tron we used something like this:
`const terms = `${name || ""} ${address}`;
  return terms.toLowerCase().includes(query.toLowerCase().trim());`